openpype/hosts/maya/plugins/inventory/connect_yeti_rig.py

```diff
@@ -0,0 +1,156 @@
+import os
+import json
+from collections import defaultdict
+
+from maya import cmds
+
+from openpype.pipeline import (
+    InventoryAction, get_representation_context, get_representation_path
+)
+from openpype.hosts.maya.api.lib import get_container_members, get_id
+
+
+class ConnectYetiRig(InventoryAction):
+    """Connect Yeti Rig with an animation or pointcache."""
+
+    label = "Connect Yeti Rig"
+    icon = "link"
+    color = "white"
+
+    def process(self, containers):
+        # Validate selection is more than 1.
+        message = (
+            "Only 1 container selected. 2+ containers needed for this action."
+        )
+        if len(containers) == 1:
+            self.display_warning(message)
+            return
+
+        # Categorize containers by family.
+        containers_by_family = defaultdict(list)
+        for container in containers:
+            family = get_representation_context(
+                container["representation"]
+            )["subset"]["data"]["family"]
+            containers_by_family[family].append(container)
+
+        # Validate to only 1 source container.
+        source_containers = containers_by_family.get("animation", [])
+        source_containers += containers_by_family.get("pointcache", [])
+        source_container_namespaces = [
+            x["namespace"] for x in source_containers
+        ]
+        message = (
+            "{} animation containers selected:\n\n{}\n\nOnly select 1 of type "
+            "\"animation\" or \"pointcache\".".format(
+                len(source_containers), source_container_namespaces
+            )
+        )
+        if len(source_containers) != 1:
+            self.display_warning(message)
+            return
+
+        source_container = source_containers[0]
+        source_ids = self.nodes_by_id(source_container)
+
+        # Target containers.
+        target_ids = {}
+        inputs = []
+        for family, containers in containers_by_family.items():
+            if family != "yetiRig":
+                continue
+
+            for container in containers:
+                target_ids.update(self.nodes_by_id(container))
+
+                maya_file = get_representation_path(
+                    get_representation_context(
+                        container["representation"]
+                    )["representation"]
+                )
+                _, ext = os.path.splitext(maya_file)
+                settings_file = maya_file.replace(ext, ".rigsettings")
+                if not os.path.exists(settings_file):
+                    continue
+
+                with open(settings_file) as f:
+                    inputs.extend(json.load(f)["inputs"])
```

Comment:
For loop here now seems redundant

```suggestion
    yeti_rig_containers = containers_by_family.get("yetiRig")
    if not yeti_rig_containers:
        self.display_warning(
            "Select at least one yetiRig container"
        )
        return

    for container in yeti_rig_containers:
        target_ids.update(self.nodes_by_id(container))

        maya_file = get_representation_path(
            get_representation_context(
                container["representation"]
            )["representation"]
        )
        _, ext = os.path.splitext(maya_file)
        settings_file = maya_file.replace(ext, ".rigsettings")
        if not os.path.exists(settings_file):
            continue

        with open(settings_file) as f:
            inputs.extend(json.load(f)["inputs"])
```